packages/@cdktf/hcl2cdk/lib/generation.ts

```diff
@@ -97,7 +174,7 @@ export const valueToTs = async (
         return t.arrayExpression(
           await Promise.all(
             unwrappedItem.map((i) =>
-              valueToTs(scope, i, path, nodeIds, scopedIds)
+              valueToTs(scope, i, `${path}[]`, nodeIds, scopedIds)
```

Comment:
Why do we need this? is this faking an array index access? Or is path a type of some sort?

Reply:
I would not call it faking. The path in general denotes where we are when we convert a value so we can lookup auxiliar type information. The extra `[]` denote that we have entered an array. This is important to know, so that if e.g. the parent type is `["array", "string"]` we now know once we went into another level of recursion that the new type we are concerned with is `"string"`